camelot/view/action_steps/application.py

```diff
@@ -131,7 +131,7 @@ def __init__( self, menu ):
         self.menu = menu
 
     def gui_run( self, gui_context ):
-        gui_context.workspace.parent().set_main_menu(self.menu)
+        gui_context.workspace.main_window.set_main_menu(self.menu)
```

Comment:
zouden we van de set_main_menu geen classmethod maken op niveau vd MainWindow, en dan doe je hier
MainWindow.set_main_menu(gui_context.workspace.parent(), self.menu)

zo heb je in MainWindow geen reference nodig naar de QObject

Reply:
Zonder referentie naar het QObject zie ik niet meteen hoe we de QActions uit het menu kunnen verbinden (Qt connect) met het MainWindow.action_triggered slot.